### ISSUE TITLE
Minimum length of phone pin differs between docs and server

### ISSUE THREAD

**Issue by yanosz** [bigbluebutton/bigbluebutton#21710]
Version: 2.7.5

On the one hand, v2.7.x-release/docs/docs/administration/customize.md instructs the users to create a dialplan having **twice** the following line:

` <action application="play_and_get_digits" data="9 9 3 30000 # conference/conf-pin.wav ivr/ivr-that_was_an_invalid_entry.wav pin \d+"/>`.

This results in play_and_get_digits requiring 9 digits for a pin at least (c.f. https://developer.signalwire.com/freeswitch/FreeSWITCH-Explained/Modules/mod-dptools/6586933/)

One the other hand, BBB 2.7.5 is creating pins having 5 digits: 
``` 
To join this meeting by phone, dial:
+49 221 677839NN
Then enter 57385# as the conference PIN number.
```

In effect, a valid pin is rejected by BBB. I don't know, if the generated PIN is to short (i.e. there's an issue generating PINs) or if the docs are invalid and the minimum size should not be 9.

Btw. I raised this on bbb-setup before (c.f. https://groups.google.com/g/bigbluebutton-setup/c/DLYcCYFp0tA)

**Comment by ben-ba** [comment]
u have to set the corrosponding var to 9:
defaultNumDigitsForTelVoice -> source: https://docs.bigbluebutton.org/development/api/#get-post-create

and yes the default value is still on 5 -> see: https://github.com/bigbluebutton/bigbluebutton/blob/v3.0.x-release/bigbluebutton-web/grails-app/conf/bigbluebutton.properties#L179

a working solution is inside the docker repo of bigbluebutton:

the dialplan:
https://github.com/bigbluebutton/docker/blob/develop/mod/freeswitch/conf/dialplan/public/dialin.xml

the defaultNumDigitsForTelVoice: 
https://github.com/bigbluebutton/docker/blob/develop/mod/bbb-web/bbb-web.properties

and a hint if you test your configuration with https://mconf.github.io/api-mate/
clear the variable voiceBridge, otherwise i couldn't set a pin with 9 digits

the configuration isn't perfect, as you mentioned the bad pin extension only check if a pin has 5 digits.

so if a pin
has 9 digits and is right -> join conference
has 9 digits and is wrong -> hangs up
has 1,2,3,4,5,6,7,8 digits -> retry it after 30 seconds , 3 attempts, if the 3. attempt is a pin with 5 digits switch to "extension conf_bad_pin"
extension "conf_bad_pin" retry it after 30 seconds and after 3. attempt hangs up, or if the 3. attempt is with 5 digits reenter the extension "conf_bad_pin", so you can be in an endless loop

in other words u can remove the extension conf_bad_pin because it doesn't make sense.

**Comment by lonesomewalker** [comment]
Can confirm, this is STILL an issue.
defaultNumDigitsForTelVoice is set to 5 by default in /usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties

**Comment by ben-ba** [comment]
yes

see the documentation for tag 3.1.x-release
https://github.com/bigbluebutton/bigbluebutton/blob/v3.1.x-release/docs/docs/administration/customize.md
still mentioned;
`<condition field="${pin}" expression="^\d{5}$">
   <action application="answer"/>
   <action application="sleep" data="1000"/>
   <action application="play_and_get_digits" data="9 9 3 30000`

also https://github.com/bigbluebutton/bigbluebutton/blob/v3.1.x-release/bigbluebutton-web/grails-app/conf/bigbluebutton.properties#L179
is still on 5
`defaultNumDigitsForTelVoice=5`
